From 5ddffb4b2eed1e12f35fd1237ec1348eb61f8472 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Tue, 16 Jul 2024 17:19:39 -0700
Subject: [PATCH] mm/damon/Kconfig: select DAMON modules to test when test
 config is set

DAMON kunit test configs default to CONFIG_KUNIT_ALL_TESTS like others.
However, the configs depend on relevant configs for building source
files implementing the functions to test.  Hence, CONFIG_KUNIT_ALL_TESTS
doesn't enable DAMON kunit test configs unless relevant DAMON source
file build configs are enabled.  Make the test configs independent of
the source files build configs, and rather let the test configs enable
the source files build configs.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index 4fe7520601dd..996c8ff67c8d 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -14,7 +14,8 @@ config DAMON
 
 config DAMON_KUNIT_TEST
 	bool "Test for damon" if !KUNIT_ALL_TESTS
-	depends on DAMON && KUNIT=y
+	depends on KUNIT=y
+	select DAMON
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON Kunit test suite.
@@ -42,7 +43,8 @@ config DAMON_PADDR
 
 config DAMON_VADDR_KUNIT_TEST
 	bool "Test for DAMON operations" if !KUNIT_ALL_TESTS
-	depends on DAMON_VADDR && KUNIT=y
+	depends on KUNIT=y
+	select DAMON_VADDR
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON virtual addresses operations Kunit test suite.
@@ -61,7 +63,8 @@ config DAMON_SYSFS
 
 config DAMON_SYSFS_KUNIT_TEST
 	bool "Test for damon debugfs interface" if !KUNIT_ALL_TESTS
-	depends on DAMON_SYSFS && KUNIT=y
+	depends on KUNIT=y
+	select DAMON_SYSFS
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON sysfs interface Kunit test suite.
@@ -91,7 +94,8 @@ config DAMON_DBGFS
 
 config DAMON_DBGFS_KUNIT_TEST
 	bool "Test for damon debugfs interface" if !KUNIT_ALL_TESTS
-	depends on DAMON_DBGFS && KUNIT=y
+	depends on KUNIT=y
+	select DAMON_DBGFS
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON debugfs interface Kunit test suite.
-- 
2.39.2

